flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/SourceWatermarkITCase.scala

```diff
@@ -305,4 +306,56 @@ class SourceWatermarkITCase extends StreamingTestBase {
     assertThat(actualWatermark).isEqualTo(expectedWatermarkOutput)
     assertThat(sink.getAppendResults.sorted).isEqualTo(expectedData.sorted)
   }
+
+  @Test
+  def testWatermarkNotMovingBack(): Unit = {
+    val data = Seq(
+      row(1, LocalDateTime.parse("2024-01-01T00:00:00")),
+      row(3, LocalDateTime.parse("2024-01-03T00:00:00")),
+      row(2, LocalDateTime.parse("2024-01-02T00:00:00"))
+    )
+
+    val dataId = TestValuesTableFactory.registerData(data)
+
+    val ddl =
+      s"""
+         | CREATE Table VirtualTable (
+         |   a INT,
+         |   c TIMESTAMP(3),
+         |   WATERMARK FOR c as c
+         | ) with (
+         |   'connector' = 'values',
+         |   'bounded' = 'false',
+         |   'scan.watermark.emit.strategy' = 'on-periodic',
+         |   'enable-watermark-push-down' = 'true',
+         |   'disable-lookup' = 'true',
+         |   'data-id' = '$dataId'
+         | )
+         |""".stripMargin
+
+    tEnv.executeSql(ddl)
+    tEnv.getConfig.set(CoreOptions.DEFAULT_PARALLELISM.key(), "1")
```

Comment:
```suggestion
    tEnv.getConfig.set(CoreOptions.DEFAULT_PARALLELISM, 1)
```

Reply:
Does not work in Scala, unfortunately. :(

Reply:
Int.box(1)